AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -3996,6 +3994,92 @@ function Invoke-AzStorageAccountADObjectPasswordRotation {
+    in the resource group myResourceGroup to Update-AzStorageAccountSetupForAES256.
+    #>
+
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact="Medium")]
+    param(
+        [Parameter(Mandatory=$true, Position=0, ParameterSetName="StorageAccountName")]
+        [string]$ResourceGroupName,
+
+        [Parameter(Mandatory=$true, Position=1, ParameterSetName="StorageAccountName")]
+        [Alias('Name')]
+        [string]$StorageAccountName,
+
+        [Parameter(Mandatory=$true, Position=0, ValueFromPipeline=$true, ParameterSetName="StorageAccount")]
+        [Microsoft.Azure.Commands.Management.Storage.Models.PSStorageAccount]$StorageAccount
+    ) 
+
+    begin {
+        Assert-IsWindows
+        Assert-IsDomainJoined
+    }
+
+    process {
+        if ($PSCmdlet.ParameterSetName -eq "StorageAccount") {
+            $StorageAccountName = $StorageAccount.StorageAccountName
+            $ResourceGroupName = $StorageAccount.ResourceGroupName
+        }
+
+        $adObject = Get-AzStorageAccountADObject -ResourceGroupName $ResourceGroupName `
+            -StorageAccountName $StorageAccountName -ErrorAction Stop
+
+        $activeDirectoryProperties = Get-AzStorageAccountActiveDirectoryProperties `
+            -ResourceGroupName $ResourceGroupName -StorageAccountName $StorageAccountName -ErrorAction Stop
+        $domain = $activeDirectoryProperties.DomainName
+
+        if (($adObject.ObjectClass -ine "computer") -or ($adObject.SamAccountName.TrimEnd("$") -ine $StorageAccountName)) {
+            Write-Verbose -Message "Removing ADObject $($adObject.DistinguishedName)"
+            Remove-ADObject -Identity $adObject.DistinguishedName -Server $domain -Confirm:$false -ErrorAction Stop
+
+            $organizationalUnitDistinguishedName = $adObject.DistinguishedName.Substring($adObject.DistinguishedName.IndexOf(',') + 1)
+
+            $message = "Join storage account '$StorageAccountName' to domain '$domain'" `
+                + " as a computer object under '$organizationalUnitDistinguishedName'"
+            Write-Verbose -Message $message
+
+            Join-AzStorageAccount -ResourceGroupName $ResourceGroupName -StorageAccountName $StorageAccountName `
+                -Domain $domain -DomainAccountType "ComputerAccount" `
+                -OrganizationalUnitDistinguishedName $organizationalUnitDistinguishedName `
+                -ADObjectNameOverride $StorageAccountName -ErrorAction Stop
+            
+            $adObject = Get-AzStorageAccountADObject -ResourceGroupName $ResourceGroupName `
+                -StorageAccountName $StorageAccountName -ErrorAction Stop
+        }
+        
+        Write-Verbose -Message "Set AD object '$($adObject.DistinguishedName)' to use AES256 for Kerberos authentication"
+        Set-ADComputer -Identity $adObject.DistinguishedName -Server $domain `
+            -KerberosEncryptionType "AES256" -ErrorAction Stop

```

Comment:
Should we consider allowing an option for "RC4,AES"? I'm not sure how customer could enable both at the same time currently.

Reply:
Since the cmdlet is for AES256, I don't see a need for RC4. If customer really wants RC4, I think it can be set up separately.